### PR TITLE
add tests for index.js

### PR DESCRIPTION
**Pull request by levithomason**
`index.js` is now tested to ensure it contains exactly all the components from `src/components`.

### CODE REVIEW

File: src/components/Confirm/Confirm.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
 import React, {Component, PropTypes} from 'react';
 import Promise from 'bluebird';
 
-import {Modal, ModalContent, ModalFooter, ModalHeader} from 'stardust';
+import Modal from 'src/components/Modal/Modal';
+import ModalContent from 'src/components/Modal/ModalContent';
+import ModalFooter from 'src/components/Modal/ModalFooter';
+import ModalHeader from 'src/components/Modal/ModalHeader';
```
Comment (levithomason) [inline src/components/Confirm/Confirm.js:7]: Importing from `stardust` was causing a failure probably due to some circular import issue.  When in the `src` directory, let's only import exact paths.

The issue was, `stardust.Confirm` was `undefined` when it should have been a `function`.

File: test/src/index-test.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  return path.basename(key).replace('.js', '');
+});
+
+describe('index.js', () => {
+  it('has a property for every component', () => {
+    _.each(componentNames, component => {
+      expect(component in stardust);
+      expect(stardust[component]).to.be.a('function');
+    });
+  });
```
Comment (levithomason) [inline test/src/index-test.js:16]: Here we test that all components in the components dir are keys on `index.js` and that their values are functions.

File: test/src/index-test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  it('only has properties which correspond to components', () => {
+    _.each(stardust, component => expect(component in componentNames));
+  });
```
Comment (levithomason) [inline test/src/index-test.js:20]: We also test that there are no extra keys by testing that each stardust key has a component in the components dir.

File: test/src/index-test.js (hunk 2 above)
Comment (athurman) [inline test/src/index-test.js:16]: awesome tests. :100:

**Comment by athurman** [comment]
:ocean: :octopus:

File: src/components/Confirm/Confirm.js (hunk 1 above)
Comment (eanplatter) [inline src/components/Confirm/Confirm.js:7]: That's disappointing! Owel.

**Comment by eanplatter** [comment]
:fallen_leaf:

File: src/components/Confirm/Confirm.js (hunk 1 above)
Comment (levithomason) [inline src/components/Confirm/Confirm.js:7]: @eanplatter keep in mind, this is only in the `/src` dir, when building stardust components that are comprised of other stardust components.  You can still import `stardust` in `/docs` and `tests`.

**Pull request merged**